actor/v2action/service_access.go

```diff
@@ -0,0 +1,223 @@
+package v2action
+
+import (
+	"fmt"
+
+	"code.cloudfoundry.org/cli/actor/actionerror"
+	"code.cloudfoundry.org/cli/api/cloudcontroller/ccv2"
+	"code.cloudfoundry.org/cli/api/cloudcontroller/ccv2/constant"
+)
+
+// EnableServiceForAllOrgs enables access for the given service in all orgs.
+func (actor Actor) EnableServiceForAllOrgs(serviceName string) (Warnings, error) {
+	var allWarnings Warnings
+
+	services, warnings, err := actor.CloudControllerClient.GetServices(ccv2.Filter{
+		Type:     constant.LabelFilter,
+		Operator: constant.EqualOperator,
+		Values:   []string{serviceName},
+	})
+
+	allWarnings = append(allWarnings, warnings...)
+	if err != nil {
+		return allWarnings, err
+	}
+
+	if len(services) == 0 {
+		return allWarnings, actionerror.ServiceNotFoundError{Name: serviceName}
+	}
+
+	servicePlans, warnings, err := actor.CloudControllerClient.GetServicePlans(ccv2.Filter{
+		Type:     constant.ServiceGUIDFilter,
+		Operator: constant.EqualOperator,
+		Values:   []string{services[0].GUID},
```

Comment:
Hey @jenspinney that all makes sense.

It's definitely a case for new work to be more conscious about receiving multiple results when one is expected.
However, with the current behaviour already in place for quite some time, I don't see how we could make this a less of a breaking change.

Does it sound feasible to try to release the work that should fail in case of duplicate service name and the `enable-service-access` refactoring in the same CLI version bump?

This won't make the things better, but at least won't have the nondeterministic legacy behaviour as part of a official CLI release with the command refactored.